lib/src/carry_save_mutiplier.dart

```diff
@@ -0,0 +1,126 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// carry_save_multiplier.dart
+// Implementation of pipeline multiplier module.
+//
+// 2023 May 29
+// Author: Yao Jing Quek <yao.jing.quek@intel.com>
+//
+
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/rohd_hcl.dart';
+
+/// A binary multiplier using carry-save technique with pipelineing.
+///
+/// Reduces calculation time and complexity by employing the carry save
+/// multiplier, which splits numbers into smaller components and performs
+/// partial multiplications on each component separately, storing the results
+/// in a compact form. The pipeline enhances performance by breaking down the
+/// multiplication process into sequential stages, allowing for concurrent
+/// execution of multiple operations.
+///
+/// The latency of the carry save multiplier is the sum of the two inputs width
+/// `a` and `b`.
+class CarrySaveMultiplier extends Multiplier {
+  /// The list of the sum from every pipeline stages.
+  late final List<Logic> _sum;
+
+  /// The list pf carry from every pipeline stages.
+  late final List<Logic> _carry;
+
+  /// The final product of the multiplier module.
+  @override
+  Logic get product => output('product');
+
+  /// The pipeline for [CarrySaveMultiplier].
+  late final Pipeline pipeline;
+
+  /// Construct a [CarrySaveMultiplier] that multiply two list of logic in
+  /// [toMultiply].
+  CarrySaveMultiplier(Logic clk, Logic reset,
```

Comment:
let's make `clk` and `reset` named arguments so it's not ambiguous and to avoid people putting `a` and `b` there.